### PR TITLE
DOC: Make API reference intro section concise

### PR DESCRIPTION
**Pull request by wla80**
Also revise the typo "in not guaranteed" to "is not guaranteed"

### CODE REVIEW

File: doc/source/api.rst [hunk 1]
```diff
@@ -6,19 +6,18 @@ API Reference
...
-methods. In general, all classes and functions exposed in the top-level
-``pandas.*`` namespace are regarded as public.
+methods. All classes and functions exposed in ``pandas.*`` namespace are public.
 
-Further some of the subpackages are public, including ``pandas.errors``,
-``pandas.plotting``, and ``pandas.testing``. Certain functions in the
-``pandas.io`` and ``pandas.tseries`` submodules are public as well (those
-mentioned in the documentation). Further, the ``pandas.api.types`` subpackage
-holds some public functions related to data types in pandas.
+Some subpackages are public which include ``pandas.errors``,
```
Comment (jreback) [inline doc/source/api.rst]: these all could be references

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/api.rst (hunk 1 above)
Comment (wla80) [reply doc/source/api.rst]: Hello @jreback , sorry, I don't understand your comment. Could you clarify what are the further changes that I need to address for this commit? Thanks a lot.

Comment (jreback) [reply doc/source/api.rst]: use ``:module:`pandas.errors` `` and so on

Comment (wla80) [reply doc/source/api.rst]: just uploaded a new fix. Please let me know if I am still doing it wrong. Thanks

Comment (jorisvandenbossche) [reply doc/source/api.rst]: @jreback we don't have module docstring pages, so I don't think such links will work

Comment (wla80) [reply doc/source/api.rst]: Hi @jorisvandenbossche , thanks for the clarification. Just revert the commit back to origin: dccfee5

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**